src/System.Private.CoreLib/shared/System/Environment.Win32.Registry.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Internal.Win32;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static string GetEnvironmentVariableFromRegistry(string variable, bool fromMachine)
+        {
+            Debug.Assert(variable != null);
+
+#if FEATURE_APPX
```

Comment:
Do you need the ifdefs here if there is Environment.NoRegistry.cs implementation?

Reply:
We have one build of CoreCLR for Windows today that dynamically checks for Windows Store apps specific behaviors in the couple of places that it differs. These dynamic checks are under `FEATURE_APPX` ifdefs.